### ISSUE TITLE
How to get both .NET 3.5 and 4.8 frameworks in the same aspnet docker image

### ISSUE THREAD

**Issue by zorpasg** [microsoft/dotnet-framework-docker#601]
I have a legacy application that is built with Web forms on .NET 4.8. It also references some old third party libraries that require .NET framework 3.5 to operate. I've been trying to containerize this applicaiton by getting the Microsoft image [mcr.microsoft.com/dotnet/framework/aspnet:3.5](https://hub.docker.com/_/microsoft-dotnet-framework-aspnet/) and installing .NET framework 4.8 in it using the [offline runtime installer ](https://dotnet.microsoft.com/download/dotnet-framework/net48) but without any success. The message I get from the installation is:
> Final Result: Installation completed successfully with success code: (0x80070BC2), "The requested operation is successful. Changes will not be effective until the system is rebooted." (Elapsed time: 0 00:01:57).

Even though the message states that the installation completed successfully, in reality it did not. Checking the installed .NET version using the powershell command:
```
Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full"
```
returns 4.7 instead of 4.8. Note that 4.7 is the framework already installed in the base image, along with 3.5.

Restarting a container is not possible so there is no way to "finalize" the installation process by restarting the container.

This issue can be replicated using the following docker file commands:
```
FROM mcr.microsoft.com/dotnet/framework/aspnet:3.5

SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

COPY ndp48-x86-x64-allos-enu.exe C:/ndp48-x86-x64-allos-enu.exe

RUN Start-Process -Wait -NoNewWindow -FilePath 'C:\ndp48-x86-x64-allos-enu.exe' -ArgumentList '/q /norestart /passive /log C:\ndp48-x86-x64-allos-enu.txt'
```

On the other hand, using the 4.8 tagged aspnet image, is also not possible because framework 3.5 is not available in that image and cannot be enabled.

Note that an sdk image is already provided with both 3.5 and 4.8 frameworks installed by default. See [mcr.microsoft.com/dotnet/framework/sdk:3.5](https://hub.docker.com/_/microsoft-dotnet-framework-sdk/). But I need a leaner runtime environment for my application, namely the aspnet image.

Has anyone managed to install both frameworks in the aspnet image? Or is Microsoft about to release such a version as they did with the sdk image?

**Comment by mthalman** [comment]
It's not clear which version of Windows you're using but you can see an example of how 4.8 is installed on Server 2016 here: https://github.com/microsoft/dotnet-framework-docker/blob/f560e047997285a135f702bb47f08b7d04322fd5/src/runtime/4.8/windowsservercore-ltsc2016/Dockerfile#L6  Otherwise, the other Windows versions are all available here for 4.8: https://github.com/microsoft/dotnet-framework-docker/tree/master/src/runtime/4.8.  Follow that as an example and see whether that resolves your issue.

**Comment by zorpasg** [comment]
Thank you @mthalman for providing those links.
The windows version where I was trying what I mentioned above and was not working as expected, is Windows Server 2019 (version 1809). Going through the links you provided and tweaking a few things around, I finally managed to install .NET framework on the aspnet:3.5 image using the following lines in my docker file:

```
# escape=`

FROM mcr.microsoft.com/dotnet/framework/aspnet:3.5

# ========== Install the .NET Framework v4.8.0 ==========
ADD ["URL_to_private_repository_containing_ndp48-x86-x64-allos-enu.exe","dotnet-framework-installer.exe"]
		
RUN powershell -Command `
        $ProgressPreference = 'SilentlyContinue'; `
    && .\dotnet-framework-installer.exe /q `
    && del .\dotnet-framework-installer.exe `
    && powershell Remove-Item -Force -Recurse ${Env:TEMP}\*
```

To be honest I am not sure why this command works as opposed to the "Start-Process" command I used above, or the `&.\ndp48-x86-x64-allos-enu.exe /q /norestart` I have also tried, but this solved my problem.
